### PR TITLE
Feat: support basic auth private helm repo

### PR DESCRIPTION
**Pull request by wangyikewxgm**
support basic-auth private helm repo

### Description of your changes

Fixes #

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: pkg/addon/versioned_registry.go [hunk 1]
```diff
@@ -131,7 +131,7 @@ func (i versionedRegistry) loadAddon(ctx context.Context, name, version string)
 		return nil, fmt.Errorf("specified version %s not exist", version)
 	}
 	for _, chartURL := range addonVersion.URLs {
-		archive, err := common.HTTPGet(ctx, chartURL)
+		archive, err := common.HTTPGetWithOption(ctx, chartURL, nil)
```
Comment (wonderflow) [inline pkg/addon/versioned_registry.go:134]: why versioned_registry don't pass the arguments?

Comment (wangyikewxgm) [reply pkg/addon/versioned_registry.go:134]: Our addon repo is public just now, we can support private repo if user want this feature later.  And this is very easy, I think we can make a good first issue to let contributor to solve.

Comment (wonderflow) [reply pkg/addon/versioned_registry.go:134]: This is not a good first issue, you should provide some design for this issue.

Comment (wangyikewxgm) [reply pkg/addon/versioned_registry.go:134]: Yes, will support this as a whole solution for user build a private addon repo with chartmuseum.

**Review by wonderflow — APPROVED** [review]

File: pkg/apiserver/rest/usecase/helm.go [hunk 2]
```diff
@@ -158,3 +183,12 @@ func flattenKey(prefix string, src map[string]interface{}, dest map[string]inter
 		}
 	}
 }
+
+func setAuthInfo(ctx context.Context, k8sClient client.Client, secretName string) (*common.HTTPOption, error) {
```
Comment (barnettZQG) [inline pkg/apiserver/rest/usecase/helm.go:187]:
```suggestion
func setAuthInfo(ctx context.Context, k8sClient client.Client, secretNameAndNamespace  types.NamespacedName) (*common.HTTPOption, error) {
```

Comment (barnettZQG) [reply pkg/apiserver/rest/usecase/helm.go:187]: Advice to provide the namespace of the secret. For example, the addon registry could use any namespace to provide a secret.

Comment (wangyikewxgm) [reply pkg/apiserver/rest/usecase/helm.go]: Reasonable. fixed

File: pkg/apiserver/rest/usecase/helm.go [hunk 3]
```diff
@@ -61,8 +63,15 @@ type defaultHelmHandler struct {
...
 
 func (d defaultHelmHandler) ListChartNames(ctx context.Context, url string, secretName string, skipCache bool) ([]string, error) {
-	// TODO(wangyikewxgm): support authority helm repo
-	charts, err := d.helper.ListChartsFromRepo(url, skipCache)
+	var opts *common.HTTPOption
+	var err error
+	if len(secretName) != 0 {
+		opts, err = helm.SetBasicAuthInfo(ctx, d.k8sClient, types2.NamespacedName{Namespace: types.DefaultKubeVelaNS, Name: secretName})
+		if err != nil {
+			return nil, err
```
Comment (barnettZQG) [inline pkg/apiserver/rest/usecase/helm.go:71]: packaging the error by `bcode`.

**Review by barnettZQG — APPROVED** [review]
lgtm

**Review by Somefive — APPROVED** [review]

**Review by leejanee — APPROVED** [review]

**Pull request merged**